### PR TITLE
Eiger Process Plugin

### PR DESCRIPTION
**Pull request by MattTaylorDLS**

### CODE REVIEW

File: frameProcessor/include/EigerProcessPlugin.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+		char encoding[11];  // String of the form "[bs<BIT>][[-]lz4][<|>]"
+		char dataType[8];	// "uint16" or "uint32" or "float32"
+	} EigerFrameHeader;
+
+  /** Processing of Eiger Frame objects.
+   *
+   * The EigerProcessPlugin class is currently responsible for receiving a raw data
+   * Frame object and parsing the header information, before splitting the raw data into
+   * the two "data" and "reset" Frame objects.
```
Comment (GDYendell) [inline frameProcessor/include/EigerProcessPlugin.h]: I guess is left over from the PercivalProcessPlugin?

File: frameProcessor/src/EigerProcessPlugin.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+namespace filewriter
+{
+  void sendMetaMessage() {
+
+  }
+
+  EigerProcessPlugin::EigerProcessPlugin()
+  {
+    // Setup logging for the class
+    logger_ = Logger::getLogger("FW.EigerProcessPlugin");
```
Comment (GDYendell) [inline frameProcessor/src/EigerProcessPlugin.cpp]: This should be FP.* now.

File: frameProcessor/src/EigerProcessPlugin.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		p2m_dims[0] = hdrPtr->shapeSizeX;
+		p2m_dims[1] = hdrPtr->shapeSizeY;
+		p2m_dims[2] = hdrPtr->shapeSizeZ;
+		frame->set_dimensions("frame", p2m_dims);
+	  } else {
+		dimensions_t p2m_dims(2);
+		p2m_dims[0] = hdrPtr->shapeSizeX;
+		p2m_dims[1] = hdrPtr->shapeSizeY;
+		frame->set_dimensions("frame", p2m_dims);
+	  }
```
Comment (GDYendell) [inline frameProcessor/src/EigerProcessPlugin.cpp]: You could remove the duplicate lines if you make the if block just

`p2m_dims.push_back(hdrPtr->shapeSizeZ);`

with the rest outside.

File: frameProcessor/src/FileWriter.cpp [hunk 4]
```diff
@@ -458,36 +458,40 @@ void FileWriter::processFrame(boost::shared_ptr<Frame> frame)
...
 
-    // Check if we have written enough frames and stop
-    if (framesWritten_ == framesToWrite_) {
-      this->stopWriting();
-    }
+      // Check if we have written enough frames and stop
+      if (framesWritten_ == framesToWrite_) {
+        this->stopWriting();
+      }
+	}
```
Comment (GDYendell) [inline frameProcessor/src/FileWriter.cpp]: Something is wrong with the diff from 473 to 494. It seems to be mostly whitespace. Is there supposed to be any change here?

File: frameProcessor/include/EigerProcessPlugin.h (hunk 1 above)
Comment (MattTaylorDLS) [inline frameProcessor/include/EigerProcessPlugin.h]: Yeah. Changed to be up to date.

File: frameProcessor/src/EigerProcessPlugin.cpp (hunk 2 above)
Comment (MattTaylorDLS) [inline frameProcessor/src/EigerProcessPlugin.cpp]: Changed.

File: frameProcessor/src/EigerProcessPlugin.cpp (hunk 3 above)
Comment (MattTaylorDLS) [inline frameProcessor/src/EigerProcessPlugin.cpp]: agreed. Changed.

File: frameProcessor/src/FileWriter.cpp (hunk 4 above)
Comment (MattTaylorDLS) [inline frameProcessor/src/FileWriter.cpp]: Yes the diff hasn't handled this very well. It is mostly whitespace, the only change being the adding of the if, causing all of the existing code to need to be indented in, and a closing bracket at the end.

File: frameProcessor/src/EigerProcessPlugin.cpp (hunk 3 above)
Comment (GDYendell) [inline frameProcessor/src/EigerProcessPlugin.cpp]: Also, why is it called p2m?

Comment (MattTaylorDLS) [inline frameProcessor/src/EigerProcessPlugin.cpp]: Changed,

**Pull request merged**